File: lucafontanili.designpatterns.java/src/main/java/com/lucafontanili/designpatterns/behavioral/mediator/MediatorClient.java
package com.lucafontanili.designpatterns.behavioral.mediator;

import java.io.IOException;

import com.lucafontanili.designpatterns.utilities.AbstractClient;
import com.lucafontanili.designpatterns.utilities.InvalidArgumentException;

public class MediatorClient extends AbstractClient {

    public static void main(String[] args) throws InvalidArgumentException, IOException {
	main(new MediatorClient());
    }

    @Override
    public void run() throws InvalidArgumentException, IOException {
	Button button = new Button();
	TextBox textBox = new TextBox();

	button.click();
	textBox.enterText(new StringBuilder(8).append("Hi!").toString());
	button.click();

    }

}

File: lucafontanili.designpatterns.python/src/behavioral/mediator/MediatorClient.py
'''
Created on Nov 12, 2016

@author: Admin
'''
from Widget import *

def main():
    button = Button()
    text_box = TextBox()
    
    button.click()
    text_box.enter_text('Hi!')
    button.click()

if __name__ == "__main__":
    main()